Neos.Utility.Arrays/Classes/PositionalArraySorter.php

```diff
@@ -267,13 +230,12 @@ protected function extractAfterKeys(array &$arrayKeysWithPosition, array $existi
      *
      * @return array an associative array where each key of $subject has a position string assigned
      */
-    protected function collectArrayKeysAndPositions(): array
+    private function collectArrayKeysAndPositions(): array
     {
         $arrayKeysWithPosition = [];
 
         foreach ($this->subject as $key => $value) {
-            // if the value was set to NULL it was unset and should not be used
-            if ($value === null) {
+            if ($value === null && $this->removeNullValues) {
```

Comment:
I admit that I added some code cleanups with the changes.. I just couldn't stand looking at this ancient coding style..
But trust me: Apart from making this class `final` and some mostly cosmetic tweaks _this_ line is the only actual change

Reply:
ich schwöre